cmd/swarmctl/node/common.go

```diff
@@ -138,6 +138,16 @@ func getNode(ctx context.Context, c api.ControlClient, input string) (*api.Node,
 		}
 
 		if l := len(rl.Nodes); l > 1 {
+			for _, n := range rl.Nodes {
+				name := n.Spec.Annotations.Name
```

Comment:
I'm not sure I understand why this is looking at `n.Spec.Annotations.Name`. It looks like the `NamePrefixes` only filter acts on `n.Description.Hostname`, so shouldn't this code only be checking `Hostname`?